File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.c
double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int i=0,j=0,k=-1;
    int marr[nums1Size+nums2Size];

    int indx;
    while(i< nums1Size&& j< nums2Size)
    {
        if(nums1[i]<nums2[j])
        {    k++;
            marr[k]=nums1[i];
            i++;
        
        }
        else 
        {   k++;
            marr[k]=nums2[j];
            j++;
        
        }
    }
    while(i<nums1Size)
    {    k++;
        marr[k]=nums1[i];
        i++;
    
    }
     while(j<nums2Size)
    {   k++;
        marr[k]=nums2[j];
        j++;
        
    }
    indx=k/2;
    if(k%2==1)
    {   
       
    return (marr[indx]+marr[indx+1])/2.0;
    
    }
    else
    {
       return marr[indx];
    
    }

}

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int s1=0;
        int s2=0;
        int indx=0;
        int[] res=new int[nums1.length+nums2.length];
        while(s1<nums1.length && s2<nums2.length)
        {
            if(nums1[s1]<=nums2[s2])
            {
                res[indx]=nums1[s1];
                indx++;
                s1++;
            }
            else
            {
                res[indx]=nums2[s2];
                indx++;
                s2++;
            }
        }
        while(s1<nums1.length)
        {
                res[indx]=nums1[s1];
                indx++;
                s1++;
        }

         while(s2<nums2.length)
        {
                res[indx]=nums2[s2];
                indx++;
                s2++;
        }
       double output;
        if(res.length%2==0)
        {
            output=(res[res.length/2-1]+res[res.length/2])/2.0;
        }
        else
        {
            output=res[res.length/2];
        }

        return output;
    }
}